code/modules/atmospherics/machinery/airalarm.dm

```diff
@@ -198,12 +204,11 @@
 		panel_open = TRUE
 		pixel_x = (dir & 3)? 0 : (dir == 4 ? -24 : 24)
 		pixel_y = (dir & 3)? (dir == 1 ? -24 : 24) : 0
+		update_icon()
 
 	if(name == initial(name))
 		name = "[get_area_name(src)] Air Alarm"
```

Comment:
While you're dealing with the new proc here transfer the other easy to transfer things like this to init